pkg/scanners/terraform/parser/resolvers/registry.go

```diff
@@ -55,12 +56,24 @@ func (r *registryResolver) Resolve(ctx context.Context, target fs.FS, opt Option
 		hostname = parts[0]
 		parts = parts[1:]
 
-		envVar := fmt.Sprintf("TF_TOKEN_%s", strings.ReplaceAll(hostname, ".", "_"))
-		token = os.Getenv(envVar)
-		if token != "" {
-			opt.Debug("Found a token for the registry at %s", hostname)
+		ascii_hostname, err := idna.ToASCII(hostname)
+		if err != nil {
+			opt.Debug("Could not convert hostname %s to a punycode encoded ASCII string so cannot find token for this registry", hostname)
 		} else {
-			opt.Debug("No token was found for the registry at %s", hostname)
+			envVar := fmt.Sprintf("TF_TOKEN_%s", strings.ReplaceAll(ascii_hostname, ".", "_"))
+			token = os.Getenv(envVar)
+
+			// Dashes in the hostname can optionally be converted to double underscores
+			if token == "" {
+				envVar = strings.ReplaceAll(envVar, "-", "__")
+				token = os.Getenv(envVar)
+			}
+
+			if token != "" {
+				opt.Debug("Found a token for the registry at %s", hostname)
+			} else {
+				opt.Debug("No token was found for the registry at %s", hostname)
+			}
```

Comment:
While we are improving this, what do you think if we refactor this logic out to a function and write a small unit test for it?

We can probably refactor most of it from the point we start parsing for `opt.Version` as that logic seems brittle.

Reply:
I have refactored my new bit out to a function and added unit tests. Could I ask to not do the refactor for the other part of it? I am not a Go programmer, and the code for getting the version is fairly complex including some web requests.